CMakeLists.txt

```diff
@@ -38,6 +35,7 @@ endfunction()
 
 # This must be done before any languages are enabled:
 if (THRUST_TOPLEVEL_PROJECT)
+  cmake_minimum_required(VERSION 3.21)
```

Comment:
The name `ThrustCompilerHacks` doesn't really make sense anymore. `ThrustSetCUDAHostCompiler` makes it clear what is being done.